accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -906,9 +906,27 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
         iter: impl Iterator<Item = (&'a Pubkey, &'a Box<AccountMapEntry<T>>)>,
         current_age: Age,
         ages_flushing_now: Age,
+        total_entries: usize,
+        max_evictions: Option<usize>,
     ) -> (CandidatesToFlush, CandidatesToEvict) {
         let mut candidates_to_flush = Vec::new();
         let mut candidates_to_evict = Vec::new();
+        let (mut remaining_to_select, mut remaining_entries) = if let Some(max) = max_evictions {
+            // Selection sampling needs an estimate of eligible entries (skip
+            // ages in the flush window and ref_count != 1). A precise count
+            // would require scanning everything, so we assume ages are uniform
+            // and ref_count != 1 is rare, and estimate the number of elegible
+            // entries with a scale off by the non-flushing ages.
+            let estimated_total_entries_exclude_skipped_ages =
+                total_entries * (256 - ages_flushing_now as usize) / 255;
```

Comment:
Please make a constant for this. In a test there is already an `AGE_MAX` (see below). Let's bump that to a file-level constant and reuse it here.

Also, why is the divide-by at 255 and not 256?

---

https://github.com/anza-xyz/agave/blob/b9b259f3ff4952580eec26aebb010e7cf5f76aba/accounts-db/src/accounts_index/in_mem_accounts_index.rs#L1719